packages/components/src/Datalist/Datalist.component.js

```diff
@@ -0,0 +1,341 @@
+				if (!this.state.groups) {
+					break;
+				}
+				event.preventDefault();
+				if (Number.isInteger(highlightedItemIndex)) {
+					// suggestions are displayed and an item has the focus : we select it
+					this.onSelect(event, {
+						itemIndex: highlightedItemIndex,
+						sectionIndex: highlightedSectionIndex,
+					});
+				} else if (this.state.value !== this.state.previousValue) {
+					// there is no focused item and the current value is not persisted
+					// we persist it
+					this.updateValue(event, this.state.value, true);
+				}
+				// resetting suggestions to close dropdown
+				this.resetSuggestions();
+				break;
+			case keycode.codes.down:
+				event.preventDefault();
+				if (!this.state.groups) {
+					// display all suggestions when they are not displayed
+					this.updateGroups();
+					this.updateSelectedIndexes(this.state.value);
+					break;
+				}
+				this.setState({
+					focusedItemIndex: newHighlightedItemIndex,
+					focusedSectionIndex: newHighlightedSectionIndex,
+				});
+				break;
+			case keycode.codes.up:
+				event.preventDefault();
+				this.setState({
+					focusedItemIndex: newHighlightedItemIndex,
+					focusedSectionIndex: newHighlightedSectionIndex,
+				});
+				break;
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * Select an item in suggestions list
+	 * @param event The select event
+	 * @param itemIndex The item index in suggestions list
+	 */
+	onSelect(event, { sectionIndex, itemIndex }) {
+		let newValue = this.state.groups[itemIndex];
+		if (this.props.multiSection) {
+			newValue = this.state.groups[sectionIndex].suggestions[itemIndex];
+		}
+		this.updateValue(event, newValue, true);
+	}
+
+	updateSelectedIndexes(value) {
+		if (this.props.multiSection) {
+			this.props.titleMap.forEach((group, sectionIndex) => {
+				const focusedIndex = group.suggestions.findIndex(item => item.name === value);
+				if (focusedIndex > -1) {
+					this.setState({
+						focusedItemIndex: focusedIndex,
+						focusedSectionIndex: sectionIndex,
+					});
+				}
+			});
+		} else {
+			const index = this.props.titleMap.findIndex(item => item.name === value);
+			this.setState({
+				focusedItemIndex: index,
+			});
+		}
+	}
+
+	/**
+	 * Set a value.
+	 * This new value can be persisted, or not. If not, it enables the ESC key to reset the value
+	 * @param event The change event
+	 * @param value The new value
+	 * @param persist Value will be persisted if true
+	 */
+	updateValue(event, value, persist) {
+		const previousValue = persist ? value : this.state.previousValue;
+
+		this.setState({
+			// setting the filtered value so it needs to be actual value
+			value: typeof value === 'object' ? value.title : value,
+			previousValue: typeof previousValue === 'object' ? previousValue.title : previousValue,
+		});
+		if (persist) {
+			let enumValue = this.props.titleMap.find(item => item.name === value);
+			if (this.props.multiSection) {
+				this.props.titleMap.forEach(group => {
```

Comment:
Please change that into `for` statement with `break` to avoid processing all groups even if you already found it